sklearn/ensemble/_hist_gradient_boosting/splitting.pyx

```diff
@@ -639,3 +650,21 @@ cdef inline Y_DTYPE_C negative_loss(
         Y_DTYPE_C hessian,
         Y_DTYPE_C l2_regularization) nogil:
     return (gradient * gradient) / (hessian + l2_regularization)
+
+cdef inline unsigned char sample_goes_left(
+        unsigned char split_on_nan,
+        unsigned char missing_go_to_left,
+        unsigned char missing_values_bin_idx,
+        X_BINNED_DTYPE_C split_bin_idx,
+        X_BINNED_DTYPE_C bin_value) nogil:
+    return (
+        (
+            split_on_nan and
+            bin_value != missing_values_bin_idx)
```

Comment:
skip line before `)`?


also maybe add a small comment like

`# if we split on nan, nans always go to right child.`